container-host-files/etc/scf/config/scripts/forward_logfiles.sh

```diff
@@ -1,175 +1,171 @@
+                \$ActionQueueType LinkedList                    # Allocate on-demand
+                \$ActionQueueFileName agg_backlog               # Spill to disk if queue is full
+                \$ActionQueueMaxDiskSpace 32m                   # Max size for disk queue
+                \$ActionQueueLowWaterMark 2000                  # Num messages. Assuming avg size of 512B, this is 1MiB.
+                \$ActionQueueHighWaterMark 8000                 # Num messages. Assuming avg size of 512B, this is 4MiB.
+                                                                # (If this is reached, messages will spill to disk until the low watermark is reached).
+                \$ActionQueueTimeoutEnqueue 0                   # Discard messages if the queue + disk is full
+                \$ActionQueueSaveOnShutdown on                  # Save in-memory data to disk if rsyslog shuts down
+                :app-name, startswith, "vcap." @${SCF_LOG_PREFIX}${SCF_LOG_HOST}:${SCF_LOG_PORT};RFC5424Format
+                :app-name, startswith, "vcap." ~                # Stop writing SCF message logs to /var/log
 	EOF
-                echo "Rsyslog forwarder: Could not create $MAIN_CONFIG in $RSYSLOG_CONF_DIR" >> $PB_OUT
+                echo "Rsyslog forwarder: Could not create ${MAIN_CONFIG} in ${RSYSLOG_CONF_DIR}" >> "${PB_OUT}"
                 exit 0
         fi
 
-        if [[ ! -f "$RSYSLOG_CONF_DIR/$MAIN_CONFIG" ]]; then
-                echo "Rsyslog forwarder: File $MAIN_CONFIG not found in $RSYSLOG_CONF_DIR" >> $PB_OUT
+        if [[ ! -f "${RSYSLOG_CONF_DIR}/${MAIN_CONFIG}" ]]; then
+                echo "Rsyslog forwarder: File ${MAIN_CONFIG} not found in ${RSYSLOG_CONF_DIR}" >> "${PB_OUT}"
                 exit 0
         fi
 }
 
 # check if more logs to be monitored by rsyslog have come into existence since the last run
 function searchTargetDir {
-        filesAdded=1
-        for file in $1/*
-        do
-                if [ -d $file ]; then
-                        ignored=false
-                        for ignore in $IGNORE_DIR
-                        do
-                                if [[ $file == $RSYSLOG_FORWARDER_WATCH_DIR/$ignore ]]; then
-                                        echo "Ignoring $file directory"
-                                        ignored=true
-                                fi
-                        done;
-                        if [ $ignored == false ]; then
-                                if searchTargetDir $file; then
-                                        filesAdded=0
-                                fi
-                        fi
+        local dir file filesAdded=1
+        local args=( "$1" )
+        for dir in ${IGNORE_DIR} ; do
+                args=( "${args[@]}" '(' -name "${dir}" -a -prune ')' -o )
+        done
+        args=( "${args[@]}" '(' -name '*.log' -a -type f -a -print0 ')' )
+        while IFS= read -r -d '' file ; do
+                TARGET_BASENAME="$(basename "$1" .log)"
```

Comment:
The `$1` should apparently be a `$file`. As is the generated file has the same name, regardless of originating logfile, causing only one job to be logged, instead of all.

Reply:
Yes, that change fixes the issue with the missing config files.